truffle/com.oracle.truffle.dsl.processor/src/com/oracle/truffle/dsl/processor/InteropProcessor.java

```diff
@@ -231,6 +240,21 @@ private void generateErrorClass(Element e, final String pkg, final String fqn, f
         }
     }
 
+    private static boolean isInner(AcceptMessage message) {
+        try {
+            Class<?> receiverType = message.receiverType();
+            System.out.println(receiverType.toString() + " is an inner class: " + receiverType.isMemberClass());
+            return receiverType.isMemberClass();
+        } catch (MirroredTypeException mte) {
+            // wow, annotations processors are strange
```

Comment:
Yes, they are. Maybe the comment could explain that this exception is thrown most of the time and one needs to use the mirrors to inspect the class.